src/System.Private.ServiceModel/tools/CertificateGenerator/CertificateManager.cs

```diff
@@ -186,6 +199,16 @@ public static X509Certificate2 CreateAndInstallLocalMachineCertificates(Certific
                 InstallCertificateToRootStore(rootCertificate);
                 InstallCertificateToMyStore(hostCert, certificateCreationSettings.ValidityType == CertificateValidityType.Valid);
                 s_localCertificate = hostCert;
+
+                // Create the PeerTrust cert
+                certificateCreationSettings = new CertificateCreationSettings()
+                {
+                    FriendlyName = "WCF Bridge - UserPeerTrustCertificateResource",
+                    Subject = fqdn,
+                    SubjectAlternativeNames = new string[] { fqdn, hostname, "localhost" }
+                };
+                var peerCert = certificateGenerator.CreateMachineCertificate(certificateCreationSettings).Certificate;
+                InstallCertificateToTrustedPeopleStore(peerCert, certificateCreationSettings.ValidityType == CertificateValidityType.Valid);
             }
```

Comment:
I don't think this helps us with the case where we have peer-only trust, right? As in, this cert will still be chain trusted because it will build up to the root cert we install to the system